lib/register.js

```diff
@@ -292,10 +292,15 @@
       else {
         module.dependencies.push(null);
       }
-
-      // run the setter for this dependency
-      if (module.setters[i])
-        module.setters[i](depExports);
+      
+      // run setters for all entries with the matching dependency name
+      var originalIndices = entry.originalIndices[i];
+      for (var j = 0, len = originalIndices.length; j < len; ++j) {
+        var index = originalIndices[j];
+        if (module.setters[index]) {
+          module.setters[index](depExports);
+        }
+      }
```

Comment:
Can we not use the same method we did to fix this in the module loader at https://github.com/ModuleLoader/es6-module-loader/commit/466616a876f3125b03e1772e78e83c2da9551d60#diff-06d63264bf4625da844047fdd2732c0eL140 here?